src/main/java/org/broadinstitute/hellbender/engine/spark/AddContextDataToReadSpark.java

```diff
@@ -31,78 +32,20 @@
-
-        // Join Reads and Variants, Reads and ReferenceBases
-        JavaPairRDD<GATKRead, Iterable<Variant>> readiVariants = JoinReadsWithVariants.join(mappedReads, variants);
-        JavaPairRDD<GATKRead, ReferenceBases> readRefBases = JoinReadsWithRefBases.addBases(referenceDataflowSource, mappedReads);
-
-        // For testing we want to know that the reads from the KVs coming back from JoinReadsWithVariants.Join
-        // and JoinReadsWithRefBases.Pair are the same reads from "reads".
-        boolean assertsEnabled = false;
-        assert assertsEnabled = true; // Intentional side-effect!!!
-        // Now assertsEnabled is set to the correct value
-        if (assertsEnabled) {
-            assertSameReads(mappedReads, readRefBases, readiVariants);
-        }
-
-        JavaPairRDD<GATKRead, Tuple2<Iterable<Iterable<Variant>>, Iterable<ReferenceBases>>> cogroup = readiVariants.cogroup(readRefBases);
-        return cogroup.mapToPair(in -> {
-            ReadContextData readContextData = null;
-            try {
-                List<Variant> lVariants = makeListFromIterableIterable(in._2()._1());
-
-                ReferenceBases refBases = Iterables.getOnlyElement(in._2()._2());
-                readContextData = new ReadContextData(refBases, lVariants);
-            } catch(NoSuchElementException e) {
-                throw new GATKException.ShouldNeverReachHereException(e);
-            }
-            return new Tuple2<>(in._1(), readContextData);
-        });
-    }
-
-    private static <T> List<T> makeListFromIterableIterable(Iterable<Iterable<T>> iterables) {
-        List<Iterable<T>> listIterableT = Lists.newArrayList(iterables);
-        List<T> listT = Lists.newArrayList();
-        if (!listIterableT.isEmpty()) {
-            final Iterable<T> iterableT = Iterables.getOnlyElement(iterables);
-            // It's possible for the iterableT to contain only a null T, we don't
-            // want to include that.
-            final T next = iterableT.iterator().next();
-            if (next != null) {
-                listT = Lists.newArrayList(iterableT);
-            }
+        // Join Reads and Variants
+        JavaPairRDD<GATKRead, Iterable<Variant>> withVariants = JoinReadsWithVariants.join(mappedReads, variants);
+        // Join Reads with ReferenceBases
+        JavaPairRDD<GATKRead, Tuple2<Iterable<Variant>, ReferenceBases>> withVariantsWithRef;
+        if (joinStrategy.equals(JoinStrategy.BROADCAST)) {
+            withVariantsWithRef = BroadcastJoinReadsWithRefBases.addBases(referenceDataflowSource, withVariants);
+        } else if (joinStrategy.equals(JoinStrategy.SHUFFLE)) {
+            withVariantsWithRef = ShuffleJoinReadsWithRefBases.addBases(referenceDataflowSource, withVariants);
+        } else {
+            throw new IllegalArgumentException("Unknown JoinStrategy");
         }
-        return listT;
-
-    }
-
-    private static void assertSameReads(final JavaRDD<GATKRead> reads,
```

Comment:
It's no longer being used.  If anything, this should be moved into a test somewhere.